### ISSUE TITLE
Block mutator

### ISSUE THREAD

**Issue by richardwerkman** [stryker-mutator/stryker-net#208]
Block statements should be mutated into empty blocks. 

For example: 
``` csharp
if (true)
{
    i = 1 + 1;
    return i;
}
```
should become:
``` csharp
if (true)
{}
```
This is a tricky issue, since these mutations can break the program. For example:
``` csharp
public int Test()
{ }
```
Will not compile since it misses it's return statement. Normally Stryker.NET will handle compile errors by rollbacking the mutation that caused the compile error. But in this case Stryker will not be able to know what mutation caused the error. We will have to find a sollution for this.

**Comment by simondel** [comment]
Can't you figure out the return value and just return the default value?

**Comment by richardwerkman** [comment]
In C7.1 we can write: `return default;` 

This solves our problem with the rollback.

**Comment by kevinlyles** [comment]
I have a [branch](https://github.com/kevinlyles/stryker-net/tree/block-mutator-test) with what I've tried so far.  I think there's a lot more we'll need to handle than just whether we have to have a return statement.  For instance, that branch fails when run on the Core project, because we remove the yield break in the first if in MethodBodyMutator.ApplyMutations, and then the variable declared in the if condition isn't present for the later code.

Another option might be to allow a mutator to flag itself as potentially compile-breaking.  Its mutations could be applied one at a time, with a compile check and (when needed) rollback to previous state after each one.  That would allow the mutator itself to be a lot simpler, and prevent errors we just didn't think of before releasing the mutator.

**Comment by kevinlyles** [comment]
Note that `return default(type);` worked just fine in older versions of C# (and is what I used in my branch).

**Comment by perdignus** [comment]
I have made the needed functionality for this, which uses `return default(TYPE)`, where type is the required return type. For void it is an empty block.
I won't make a pr just yet, because the code needs some cleaning and tests.

**Comment by richardwerkman** [comment]
Hi @kevinlyles @perdignus please comment here before you start any work. It seems you both worked on the same issue. Maybe you can combine your work if needed?

**Comment by kevinlyles** [comment]
> I'd like to play around with this idea a bit.

I did, two weeks ago.

**Comment by richardwerkman** [comment]
Oh I interpreted that as playing around with the default return values. Anyway I'm looking forward to the PR :)

**Comment by perdignus** [comment]
no problem, I mainly did it to get to know the program better so that I can contribute more in the future

**Comment by richardwerkman** [comment]
@kevinlyles @perdignus Has any of you had time to look at this? If not, we would like someone else to pick this up.

**Comment by rouke-broersma** [comment]
Implemented by #1717
